### DISCUSSION TITLE
把表格结构预测换成tablemaster出错

### DISCUSSION THREAD

**Discussion by mingxingpu** [PaddlePaddle/PaddleOCR#13006] [Q&A]
您好！ 我使用如下命令       
python table/predict_table.py 
--det_model_dir=inference/ch_ppocr_server_v2.0_det_infer 
--rec_model_dir=inference/ch_ppocr_server_v2.0_rec_infer 
--table_model_dir=inference/table_structure_tablemaster_infer 
--rec_char_dict_path=../ppocr/utils/ppocr_keys_v1.txt 
--table_char_dict_path=../ppocr/utils/dict/table_master_structure_dict.txt 
--image_dir=../../images/table_images/table.png 
--output=../output/table --table_max_len=480                                    
          报错   
NotFoundError: Operator (logical_and) does not have kernel for data_type[float]:data_layout[ANY_LAYOUT]:place[CUDAPlace(0)]:library_type[PLAIN].
      [Hint: Expected kernel_iter != kernels.end(), but received kernel_iter == kernels.end().] (at /paddle/paddle/fluid/framework/operator.cc:1276)
      [operator < logical_and > error]
      请问如何解决呢。

**Comment by cuicheng01** [comment]
您好，麻烦提供下完整的环境信息呢

**Comment by mingxingpu** [comment]
@cuicheng01 
CUDA_VERSION=11.4     
我在conda中完成的环境配置：conda create -n paddle_env python=3.8  &&  conda activate paddle_env
然后安装paddlepaddle-gpu：conda install paddlepaddle-gpu==2.6.0 cudatoolkit=11.2 -c https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/Paddle/ -c conda-forge 
然后安装paddleocr：pip3 install "paddleocr>=2.6.0.3"

然后参照https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/ppstructure/table/README_ch.md#41-%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B的方式运行python脚本，并把相关模型文件放到正确的目录。其中
检测模型是ch_PP-OCRv3_det_infer
识别模型是ch_PP-OCRv3_rec_infer
表结构预测模型是table_structure_tablemaster_infer


然后这次执行的命令：
python table/predict_table.py \
--det_model_dir=inference/ch_PP-OCRv3_det_infer \
--rec_model_dir=inference/ch_PP-OCRv3_rec_infer \
--table_model_dir=inference/table_structure_tablemaster_infer \
--rec_char_dict_path=../ppocr/utils/ppocr_keys_v1.txt \
--table_char_dict_path=../ppocr/utils/dict/table_master_structure_dict.txt \
--image_dir=../../images/table_images/table.png \
--output=../output/table \
--table_max_len=480 \
--table_algorithm=TableMaster

没有报错了，代码顺利运行。但是效果比较差。
输入的图片是
![Uploading table.png…]()
生成的结果图是
![table_re](https://github.com/PaddlePaddle/PaddleOCR/assets/59463530/0712d728-86da-4499-890c-14aa7d27cefa)
生成的表格如图
<img width="809" alt="table_xlsx" src="https://github.com/PaddlePaddle/PaddleOCR/assets/59463530/bc5013e7-e914-4020-9f98-79224d3e2e6e">

请问怎么能够提升效果呢，有没有哪些参数可以调整的？

**Comment by Leezzzzz** [comment]
请问你的预测的时候用的时Linux还是Windows系统
